webapp/src/ts/services/transitions.service.ts

```diff
@@ -13,11 +14,13 @@ export class TransitionsService {
     private settingsService:SettingsService,
     private validationService:ValidationService,
     private mutingTransition:MutingTransition,
+    private userReplaceTransition:UserReplaceTransition,
   ) {
   }
 
   private readonly AVAILABLE_TRANSITIONS = [
-    { name: 'muting', transition: this.mutingTransition }
+    { name: 'muting', transition: this.mutingTransition },
+    { name: 'user_replace', transition: this.userReplaceTransition },
```

Comment:
I liked using the readonly property `name` of the transitions, was it causing issues?

Reply:
So, honestly, this caused some of the unit tests to fail and I reverted it which fixed the problem, but I did not do any more investigation.  

Now I have looked at this closer and I see the test failures were just because the mocks in the tests were not mocking the `name` property on the transition (easy fix).  I fixed that and now have refactored this code to actually just use that property directly...